sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -191,22 +196,56 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
     cancelledDate: Date,
     newIssuingProvince: string,
   ): Promise<UpdateResult> {
-    if (!cancelledDate || !newIssuingProvince) {
-      throw new Error(
-        "Not all required fields to update a received MSFAA record were provided.",
+    return this.dataSource.transaction(async () => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      if (!cancelledDate || !newIssuingProvince) {
+        throw new Error(
+          "Not all required fields to update a received MSFAA record were provided.",
+        );
+      }
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
+        {
+          msfaaNumber: msfaaNumber,
+          cancelledDate: IsNull(),
+          newIssuingProvince: IsNull(),
+        },
+        {
+          cancelledDate: getISODateOnlyString(cancelledDate),
+          newIssuingProvince,
+        },
       );
-    }
-
-    return this.repo.update(
-      {
-        msfaaNumber: msfaaNumber,
-        cancelledDate: IsNull(),
-        newIssuingProvince: IsNull(),
-      },
-      {
-        cancelledDate: getISODateOnlyString(cancelledDate),
-        newIssuingProvince,
-      },
-    );
+      // Get the student associated with the cancelled MSFAA record.
+      const msfaaRecord = await this.repo.findOne({
+        select: {
+          id: true,
+          student: {
+            id: true,
+            user: { id: true, firstName: true, lastName: true, email: true },
+          },
+        },
+        relations: {
+          student: { user: true },
+        },
+        where: {
+          msfaaNumber,
+        },
+      });
+      const systemUser = await this.systemUsersService.systemUser();
+      // Only if the msfaa record was successfully cancelled, then save the msfaa cancellation notification.
+      // This also takes care of the scenario where the msfaa record was already cancelled before in which case the record will not be update and hence no notification will be saved.
+      if (updateResult.affected) {
+        this.notificationActionsService.saveMSFAACancellationNotification(
```

Comment:
This method also expected the `entityManager` to be part of an existing transaction. Please check the other methods using something like `this.notificationActionsService.saveSomething` to see how they were implemented.